shell/browser/atom_autofill_driver_factory.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2019 GitHub, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/atom_autofill_driver_factory.h"
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "base/bind.h"
+#include "base/callback.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/web_contents.h"
+#include "shell/browser/atom_autofill_driver.h"
+
+namespace electron {
+
+namespace {
+
+std::unique_ptr<AutofillDriver> CreateDriver(
+    content::RenderFrameHost* render_frame_host) {
+  return std::make_unique<AutofillDriver>(render_frame_host);
+}
+
+}  // namespace
+
+AutofillDriverFactory::~AutofillDriverFactory() {}
+
+// static
+void AutofillDriverFactory::BindAutofillDriver(
+    mojom::ElectronAutofillDriverAssociatedRequest request,
+    content::RenderFrameHost* render_frame_host) {
+  content::WebContents* web_contents =
+      content::WebContents::FromRenderFrameHost(render_frame_host);
+  if (!web_contents)
+    return;
+
+  AutofillDriverFactory* factory =
+      AutofillDriverFactory::FromWebContents(web_contents);
+  if (!factory)
+    return;
+
+  AutofillDriver* driver = factory->DriverForFrame(render_frame_host);
+  if (driver)
+    driver->BindRequest(std::move(request));
+}
+
+AutofillDriverFactory::AutofillDriverFactory(content::WebContents* web_contents)

```

Comment:
Is it necessary to eagerly create drivers for all living frames ? `BindAutofillDriver` seems to be called on demand, would simplify this class if the drivers were lazy loaded.